firestore-bigquery-export/functions/src/index.ts

```diff
@@ -51,6 +61,22 @@ exports.fsexportbigquery = functions.firestore
       const changeType = getChangeType(change);
       const documentId = getDocumentId(change);
 
+      if (eventChannel) {
+        await eventChannel.publish({
+          type: `com.bigqueryexport.v1.sync.start`,
+          data: {
+            before: {
+              ...change.before,
```

Comment:
Is this safe to do? It'll try to seralize everything in the snapshot which could throw an error or be unreadable. Would be safer to have an explicit object of data imo.

Reply:
Yeah, we probably don't need that data. Surely, the context and data would be enough?

```js
data: {
            before: {
              data: change.before.data(),
            },
            after: {
              data: change.after.data(),
            },
            context,
          },
```

Reply:
maybe the id too?